### PR TITLE
add FairDummies method

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: ethicml/algorithms/inprocess/fair_dummies.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from typing_extensions import TypedDict
+
+from ranzen import implements
+
+from ethicml.algorithms.inprocess.in_algorithm import HyperParamType
+from ethicml.algorithms.inprocess.in_subprocess import InAlgorithmSubprocess
+
+
+class FairDummiesArgs(TypedDict):
+    """Args for the Agarwal implementation."""
```
Comment (thomkeh) [inline ethicml/algorithms/inprocess/fair_dummies.py:18]:
```suggestion
    """Args for the Fair dummies implementation."""
```

File: ethicml/implementations/fair_dummies_modules/model.py [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+
+
+class PandasDataSet(TensorDataset):
+    """Pandas based dataset."""
+
+    def __init__(self, *dataframes: pd.DataFrame):
+        tensors = (self._df_to_tensor(df) for df in dataframes)
+        super().__init__(*tensors)
+
+    def _df_to_tensor(self, df: pd.DataFrame) -> torch.Tensor:
```
Comment (thomkeh) [inline ethicml/implementations/fair_dummies_modules/model.py:36]:
```suggestion
    def _df_to_tensor(self, df: Union[pd.DataFrame, pd.Series]) -> torch.Tensor:
```

**Review by thomkeh — APPROVED** [review]

File: ethicml/algorithms/inprocess/kamiran.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def _train(
         C: float,
         kernel: Optional[KernelType],
         seed: int,
-    ) -> sklearn.linear_model._base.LinearModel:
+    ) -> ethicml.implementations.pytorch_common.LinearModel:
```
Comment (thomkeh) [inline ethicml/algorithms/inprocess/kamiran.py:87]: I'm guessing PyCharm renamed this when you refactored, but this seems wrong.

File: ethicml/data/tabular_data/nursery.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            class_label_spec = "class_not_recom"
+            class_label_prefix = ["class"]
+        else:
+            raise NotImplementedError
+
+        name = f"Nursery {self.split.value}"
+
+        super().__init__(
+            name=name,
+            num_samples=45222,
```
Comment (olliethomas) [inline ethicml/data/tabular_data/nursery.py:74]:
```suggestion
            num_samples=12960,
```

**Pull request merged**